src/app/services/geo.service.ts

```diff
@@ -106,8 +106,20 @@ export class GeoService {
               data.d.coordinates.longitude,
             ],
           } as GeoAddress;
-        })
-      )
+        });
+
+        // add whole city
+        if (!isSearchPostal && all && all.length > 0) {
+          all.unshift({
+            city: all[0].city + '',
+            postalcode: '',
+            coordinates: all[0].coordinates,
```

Comment:
Will this just take the center of the first postal? So the coordinates are probably not in the real center of the city, right?

Reply:
yes, that's right. In our geodata the coordinates for the cities or for the centre of a city are missing. But it is not really bad, because I have implemented a logic which increases the radius from 5 to 25 km as soon as you have selected a whole city. Together with the sorting by distance and the radius slider, we have solved the most serious problems. 

The missing city/coordinates can be updated in the next release, then we have an even better accuracy.